src/opower/utilities/base.py

```diff
@@ -14,6 +14,11 @@ def __init_subclass__(cls, **kwargs) -> None:
         super().__init_subclass__(**kwargs)
         cls.subclasses.append(cls)
 
+    @staticmethod
+    def requires_mfa() -> bool:

```

Comment:
> Is your plan to only show the 3rd input textbox in the config flow if this function returns true? Or always show the textbox and validate it is set when this returns true?

Yeah I was just always going to show it with an (optional) label or something. Introducing a second step to configuration seems a little annoying (and we'd have to dump & restart the auth flow for it after we detect the requirement).

>  A future use case to consider is that for the Exelon utilities you can enable/disable 2FA so I foresee the mfa_secret being optional for them so "requires" won't be quite right for them.

Yeah, the "requires" is more of a code reference, as in "the constructor requires providing the argument". Although it could also be used for validation purposes (but not both). If we add the arg to everything, this method could go away/become a UX-validation-only thing.

Reply:
I think it would be good to keep the requires_mfa but rename it to accepts_mfa.
It would also be good to add a parameter accepts_mfa or supports_mfa or similar in get_supported_utilities and default it to false. If it's true filter out any utilities, such as coned, that return true for their accepts_mfa. Then in your core PR you can call get_supported_utilities with True. Same from demo.py. The advantage is we will be able to continue bumping the version of this library for any fixes in other utilities and it will be backwards compatible. Otherwise we are risking prematurely releasing conEdison as an option in the config flow before the integration can specify the secret.